### DISCUSSION TITLE
Loss of context happens - Each time I open the IDE, Piece uses a new chat (contextless)

### DISCUSSION THREAD

**Discussion by LeOndaz** [pieces-app/support#147] [General]
Steps to reproduce:

- Use pieces in VScode, the chat appears in pieces as expected
- Both two way messaging works

At first, pieces which is opened by (Quick fix) doesn't have the project context, only the code sent, a bit not expected but fine

The main problem is that if you reload VSCode, and try again, Pieces uses a new chat, so each time I have to tell it about the project and repeat myself

What can we do about that?

**Comment by caleb-at-pieces** [comment]
For a short term fix on this I am definitely willing to add a setting where certain actions i.e the pieces fix action don't create a new conversation every time.

Long term though I do have plans on being able to use the entire project as context during a fix action, which will hopefully improve the accuracy of the generation.
